File: lib/java-stable_old2/m/sql/StreamOutput.java
package m.sql;

public interface StreamOutput
{
  public void close() throws Exception;
  public void flush() throws Exception;
  
  public void write(int b) throws Exception;
  public void write(byte[] buffer) throws Exception;
  public void write(byte[] buffer, int length) throws Exception;
  public void write(byte[] buffer, int offset, int length) throws Exception;
}

File: cpp/sunaptos/bak/sunaptos_45/src/_core/stream/streamoutput.cpp
#include "streamoutput.h"

//StreamOutput& StreamOutput::operator<<(number e)
//{
//  Buffer b;
//  b.set(&e, sizeof(number));
//  write(b);
//  return *this;
//}

StreamOutput& StreamOutput::operator<<(const Buffer& b)
{
  write(b);
  return *this;
}
